src/state/events/reducers.js

```diff
@@ -106,37 +107,37 @@ const eventReducer = (state = initialState, action) => {
           return ele
         })
       }
-    case REQUEST_EVENTS_COUNTS_SUCCESS:
+    case REQUEST_FEDERAL_EVENTS_COUNTS_SUCCESS:
       return {
         ...state,
-        eventsCounts: action.payload,
+        eventsCounts: {
+          ...state.eventsCounts,
+          federal: action.payload,
+        }
       }
-    case REQUEST_TOTAL_EVENTS_COUNTS_SUCCESS:
+    case REQUEST_STATE_EVENTS_COUNTS_SUCCESS:
       return {
         ...state,
-        totalEventsCounts: action.payload
+        eventsCounts: {
+          federal: state.eventsCounts.federal,
+          ...action.payload,
```

Comment:
this looks fishy to me. What are you trying to do?

Reply:
Typically I would put something like:

`        eventsCounts: {
          ...state.eventsCounts,
          ...action.payload,
        }`

...so that when updating the state counts, the federal counts would be unchanged. The problem is that if there's an event in a state, and someone else approves/deletes it, the reducer will not update the count for that state to 0, because the key will no longer exist in firebase & it won't be in the payload either. The previous count in the property spread will take precedence and remain unchanged. 

The code the way I have it is basically saying 'when updating the state counts, keep only the federal counts unchanged, and completely overwrite all of the state counts data.'

Reply:
oh, if i understand you right, I think what would be clearer is:
```
eventCounts: {
...state.eventCounts,
state: action.payload.state,
}```
also for clarity, i tried to use `usState` instead of `state` in redux

Reply:
I like the usState idea, having 'state' everywhere has been bugging me too. 

The trouble is, as-is the eventCounts keys match the FEDERAL_STATE_RADIO_BUTTONS array, they're all flat, at the same level. If I put all the usStates inside another nested object, I'll just have to flatten them out again immediately so they can be consumed by the FederalStateRadioSwitcher. 

I did think about incorporating the FEDERAL_STATE_RADIO_BUTTONS constant into the logic that pulls the usState counts from firebase, so it would check for each usState defined in the constant and if it doesn't see it default to zero. That might actually be a little more robust, but at the time it seemed like overkill.